pkg/api/handlers/containers_create.go

```diff
@@ -36,9 +50,21 @@ func CreateContainer(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	cc.Name = query.Name
 	var pod *libpod.Pod
 	ctr, err := shared.CreateContainerFromCreateConfig(runtime, &cc, r.Context(), pod)
 	if err != nil {
+		if strings.Contains(err.Error(), "invalid log driver") {
+			// this does not quite work yet and needs a little more massaging
+			w.WriteHeader(http.StatusInternalServerError)
```

Comment:
WriteHeader() should be called right before the response body.  Otherwise, your additional headers won't be written.